src/connectors/infinite-hits/connectInfiniteHits.ts

```diff
@@ -262,30 +262,40 @@ const connectInfiniteHits: InfiniteHitsConnector = (
       },
 
       getWidgetState(uiState, { searchParameters }) {
-        const page = searchParameters.page || 0;
-
-        if (!hasShowPrevious || page === 0 || page + 1 === uiState.page) {
+        if (!hasShowPrevious) {
           return uiState;
         }
 
+        // The page in the UI state is incremented by one
+        // to expose the user value (not `0`).
+        const page = (searchParameters.page || 0) + 1;
```

Comment:
We're not suppose to sync the zero value (see [`pagination`](https://github.com/algolia/instantsearch.js/blob/f69ea2614a9d92fb48ba5535fae11537beb7818d/src/connectors/pagination/connectPagination.js#L170-L179)).